packages/features/calAIPhone/providers/retellAI/services/CallService.ts

```diff
@@ -165,6 +165,93 @@ export class CallService {
     };
   }
 
+  async createWebCall({
+    agentId,
+    userId,
+    teamId,
+    timeZone,
+    eventTypeId,
+  }: {
+    agentId: string;
+    userId: number;
+    teamId?: number;
+    timeZone: string;
+    eventTypeId: number;
+  }) {
+    if (!agentId?.trim()) {
+      throw new HttpError({
+        statusCode: 400,
+        message: "Agent ID is required and cannot be empty",
+      });
+    }
+
+    await this.validateCreditsForTestCall({ userId, teamId });
+
+    await checkRateLimitAndThrowError({
+      rateLimitingType: "core",
+      identifier: `web-call:${userId}`,
+    });
```

Comment:
I think we shouldn't check for limits in the services. Limits could be different for webapp and API V2, so limits should be checked in the controller or trpcHandler itself.